daemon/create.go

```diff
@@ -144,5 +144,9 @@ func (daemon *Daemon) VolumeCreate(name, driverName string, opts map[string]stri
 	if err != nil {
 		return nil, err
 	}
+
+	if driverName != "" && v.DriverName() != driverName {
```

Comment:
Yep, definitely needed but we need to get the default volume driver if it is `""` instead of ignoring.